alainmucyo/ims-backend
# Ft user login to the system

#### What does this PR do?
Allows a user to sign In using an email and password

#### Description of Task to be completed?

- Check if credentials are correct
- Validate the inputs
- Provide a user a JWT token on successful sign In. 
- Show errors if there is any

#### How should this be manually tested?

How should this be manually tested?
- Clone the repo `git clone https://github.com/alainmucyo/ims-backend.git`
- Switch your working directory `cd ims-backend`
- Checkout the branch `git checkout ft-user-login`
- Create a` .env file` in the root folder and update it using data found in `env-example`
- Install all dependencies `npm install`
- Run migrations `npm run migrate`
- Run seeds `npm run seed`
- Start the server `npm run dev`
- Using postman on Login endpoint `api/auth/login` login by providing email and password in the request body
- You should see a successful login message and the returned token if it's successful or errors if not successful

#### Any background context you want to provide?

> As there is no signup, for the correct credentials you have to use the information in the seeders user file(`path:src/database/seeders/20210216134513-User.js`)

#### What are the relevant trello stories?
[Story](https://trello.com/c/QnLbs2eW)

#### Screenshots (if appropriate)

![login success](https://user-images.githubusercontent.com/58434968/108111034-d2df7100-709c-11eb-9fd2-8abd83299ebb.PNG)
![login wrong credentials](https://user-images.githubusercontent.com/58434968/108111060-d96de880-709c-11eb-861e-61db103c4185.PNG)

#### Questions:

N/A
